src/components/Home.tsx

```diff
@@ -0,0 +1,59 @@
+import React from "react";
+import { connect, ConnectedProps } from "react-redux";
+import { Link } from "react-router-dom";
+import HabitCard from "./HabitCard";
+import { getHabitIds } from "../utils";
+import { State } from "../store/types";
+
+const mapState = (state: State) => {
+  const user = state.currentUser;
+  return user
+    ? { habitIds: getHabitIds(state.userStates[user].habits) }
+    : { habitIds: [] };
+};
+
+const connector = connect(mapState);
+
+type PropsFromRedux = ConnectedProps<typeof connector>;
+
+type Props = PropsFromRedux & {
+  collapsed: boolean;
+};
+
+class Home extends React.Component<Props> {
+  habitList() {
+    return this.props.habitIds.map((id) => {
+      return (
+        <div key={id}>
+          <HabitCard collapsed={this.props.collapsed} id={id} />
+        </div>
+      );
+    });
+  }
+
+  renderEmpty = () => {
+    return (
```

Comment:
It's a bit odd seeing both the `method() { ... }` syntax and `method = () => { ... }` syntax mixed together in this file (and elsewhere). It would be nice to standardize on one or the other. I think the arrow syntax, the option used for `renderEmpty`, is the more common one used in the code base so it would probably be easier to standardize on that.

Reply:
If we do standardize on arrow functions, then both `habitList() { ... }` and `render() { ... }` would be rewritten as `habitList () => { ... }` and `render () => { ... }`.